### PR TITLE
Fixed surface locking for shadow server.

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

File: server/shadow/Mac/mac_shadow.c [hunk 1]
```diff
@@ -381,14 +382,16 @@ static void (^mac_capture_stream_handler)(
...
 	  return;
 
+  EnterCriticalSection(&(surface->lock));
   mac_shadow_capture_get_dirty_region(subsystem);
   surfaceRect.left = 0;
   surfaceRect.top = 0;
   surfaceRect.right = surface->width;
   surfaceRect.bottom = surface->height;
   region16_intersect_rect(&(surface->invalidRegion), &(surface->invalidRegion), &surfaceRect);
+  empty = region16_is_empty(&(surface->invalidRegion); LeaveCriticalSection(&(surface->lock));
```
Comment (mfleisz) [inline server/shadow/Mac/mac_shadow.c:392]: Formatting (the following if looks off too ... wondering how this got through the checker)

Comment (akallabeth) [reply server/shadow/Mac/mac_shadow.c:392]: oh, that is not the formatting...

**Pull request merged**